### PR TITLE
Replace colorize with rainbow

### PR DESCRIPTION
**Pull request by robertino**
This change replaces `colorize` gem with `rainbow` gem.
That is because `fasterer` has MIT license, and `colorize` has GNU GPL license, hence `rainbow` with its MIT license seems a better fit.

### CODE REVIEW

**Comment by DamirSvrtan** [comment]
Hi @robertino - thank you for your PR! Can you tell me a bit more about the motivation behind this PR?

**Comment by robertino** [comment]
Hi! The motivation was purely to use one license type. Considering fasterer has MIT defined, it would make sense to support it with another library with the same license and equivalent functionality, rather that having more restrictive one. Some projects could oppose using library with GNU GPL which is requirement of fasterer right now, while they would be fine with MIT. Hence suggested swap of colorize for rainbow.

**Comment by DamirSvrtan** [comment]
Hi @robertino - thanks for the explanation.

Is this a hypothetical situation that you are describing, or an actual situation you or @salzig have run into?

**Comment by robertino** [comment]
Hi,
This was an actual issue we've had as an organisation.
For the commercial product we were building we were making sure we don't have any dependencies on GPL code.

**Comment by huerlisi** [comment]
I was about to provide a similar MR, happy to find this one:-) 

@DamirSvrtan I actually have to drop `fasterer` for now from our code base.

(As usual when making such statement: IANAL :-)
As @robertino stated the GPL is a more restrictive (or freedom preserving, depending on view:-) license that MIT. The GPL license family is so called copyleft and requires linked code to fall under the same or a compatible license. This means that a codebase that uses `colorize` would need to follow the rules of the GPL and therefore the source would need to be made available. 

This is also the case when the dependency is not direct, but goes through another library like `fasterer`. This is especially tricky as developers often only check the licenses of libraries they add, but not the sub-dependencies. So if you, for whatever reason, decide to stay with `colorize` it would be great to explicitly mention that it uses a GPL'd dependency.

The whole topic is not so relevant for many as `fasterer` is a library mostly used in development. And the GPL allows almost unrestricted local use, as long as you don't distribute. It's therefore fine to use a GPL dependency for non-OSS tools as long as it does not end up in the distributed product.

Hope these explanations help a bit. And want to say a bit "Thank You!" for this gem!

File: bin/fasterer [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env ruby
-$LOAD_PATH << File.expand_path('../../lib', __FILE__)
+$LOAD_PATH << File.expand_path('../lib', __dir__)
```
Comment (huerlisi) [inline bin/fasterer:2]: I think this is not related to the main topic of this change, right?

Comment (robertino) [reply bin/fasterer:2]: You are right, it is not. It's a tiny change suggested by rubocop, I couldn't resist adding this bit, thinking of it as undeserving of its own PR.

Comment (huerlisi) [reply bin/fasterer:2]: Sure, and it's no my call to decide this;-)

It just kind of looked like it could be something you needed to adapt to get it working locally so I wanted to hint about this.

**Comment by robertino** [comment]
You could compare this PR to the following release :) https://weblog.rubyonrails.org/2021/3/26/marcel-upgrade-releases/

**Comment by DamirSvrtan** [comment]
I have replaced colorize with an internal implementation, so I'm closing this PR. Thanks everyone for the comments and the contributions!
